content/components/only-manipulate-the-dom-via-the-renderer.md

```diff
@@ -0,0 +1,52 @@
+---
+title: only manipulate the DOM via the Renderer
+author: 
+  name: Billy Lando
+  url: https://github.com/billyjov
+---
+
+# Problem 
+
+According to the Angular documentation, relying on direct DOM access creates tight coupling between your application and rendering layers which will make it impossible to separate the two and deploy your application into a web worker. Permitting direct access to the DOM can make your application more vulnerable to **XSS** attacks. Also, using `document` object is not recommended as it's not available on other renderer specific platforms like a server side rendering and will not work when using **Ahead-of-Time** compilation.
+
+# Solution 
+
+Always try to prefer the `Renderer2` for DOM manipulations. It provides an API that can safely be used even when direct access to native elements is not supported.
+
+```ts
+@Component({
+  ...
+})
+export class SomeComponent implements OnInit {
+
+  constructor(private elementRef: ElementRef) {}
+    
+  ngOnInit() {
+    this.elementRef.nativeElement.style.backgroundColor = '#fff';
+    this.elementRef.nativeElement.style.display = 'inline';
+    const textareaElement = document.querySelector('textarea');
+  }
+}
+```
+
+We can refactor this by using a combination of `ElementRef` and `Renderer2`:
+
+```ts
+@Component({
+  ...
+})
+export class SomeComponent implements OnInit {
+
+  constructor(private elementRef: ElementRef, private renderer: Renderer2) {}
+    
+  ngOnInit() {
+    this.renderer.setElementStyle(this.elementRef.nativeElement, 'backgroundColor', '#fff');
+    this.renderer.setElementStyle(this.elementRef.nativeElement, 'display', 'inline');
+    const textareaElement = this.renderer.selectRootElement('textarea');
```

Comment:
I would replace this with a `ViewChild()` instead. Not sure how others feel about this but using `ViewChild()` is a more idiomatic way of querying elements from the view, such as a `textarea`.

I'd do the same for the jQuery example that we introduce above.